drivers/place/attendee_scanner.cr

```diff
@@ -0,0 +1,49 @@
+require "placeos-driver"
+require "json"
+
+class Place::AttendeeScanner < PlaceOS::Driver
+  descriptive_name "PlaceOS Room Events"
+  generic_name :AttendeeScanner
+
+  getter systems : Hash(String, Array(String)) { get_systems_list.not_nil! }
+
+  # Grabs the list of systems in the building
+  def get_systems_list
+    building_id = system[:Location].building_id.get
+    system["StaffAPI"].systems_in_building(building_id).get.as_h.transform_values(&.as_a.map(&.as_s))
+  rescue error
+    logger.warn(exception: error) { "unable to obtain list of systems in the building" }
+    nil
+  end
+
+  # Lists external guests based on email domains
+  def list_external_guests
+    guests = [] of Hash(String, JSON::Any)
+
+    systems.each do |level_id, system_ids|
+      system_ids.each do |system_id|
+        sys = system(system_id)
+        if sys.exists?(:Booking)
+          bookings = sys[:Booking][:bookings].as_a
+
+          bookings.map do |booking|
+            unless booking["host"].as_s.includes?("place.technology")
+              event_start = Time.unix(booking["event_start"].as_i)
+              current_date = Time.local
+
+              if event_start.year == current_date.year &&
+                 event_start.month == current_date.month &&
+                 event_start.day == current_date.day
+                booking["attendees"].as_a.map do |attendee|
+                  guests.push(attendee.as_h)
```

Comment:
Where there are external meetings, we want to save system.id, meeting.id, meeting uuid and an array of external attendees

Then we want to check if for each of these meetings those attendees have been invited to the building, providing the event id and UUID
* https://placeos.github.io/staff-api-swagger-ui/#/Bookings/Bookings_index
![image](https://github.com/user-attachments/assets/4af85d74-a513-4fd8-a6d9-20bc6db3e95e)

If there are no visitor bookings then need to create them using (passing in the event details as parameters)
https://placeos.github.io/staff-api-swagger-ui/#/Bookings/Bookings_create
Also probably worth discussing with Alex his flow for visitor bookings to make sure it matches

Reply:
@MrYuion what are your thoughts?

Reply:
I'm guessing this is for creating visitor bookings from calendar events made outside of our UIs.
Sounds about right in that case. 
The current UIs create a booking for each external attendee in the calendar event then links those bookings with that original calendar event.
I'm guessing you'll have to check to make sure the frontend hasn't created a booking for that attendee + event already.

Reply:
Had a bit of a trouble dealing with this but managed to get somewhere, please provide what I should put in the create_booking function.